bundle/config/mutator/resolve_variable_references.go

```diff
@@ -205,3 +222,43 @@ func (m *resolveVariableReferences) resolveOnce(b *bundle.Bundle, prefixes []dyn
 
 	return hasUpdates, diags
 }
+
+// If includeResources is true, the 'resources' config is mutation, otherwise, it is excluded.
+func (m *resolveVariableReferences) selectivelyMutate(b *bundle.Bundle, fn func(value dyn.Value) (dyn.Value, error)) error {
+	return b.Config.Mutate(func(root dyn.Value) (dyn.Value, error) {
+		rootMap := root.MustMap()
```

Comment:
In general we avoid use Must* to ensure we never panic (incidents happened). 

It's okay to log an error and and return early or even just return early if you're sure it will never happen.

Reply:
If `root` isn't a map, it violates a core assumption about how the bundle is structured. So I thought that crashing ASAP makes more sense. I can change it, but it's just more boilerplate for the code that will never be executed.

Reply:
that's okay if it's a bit more boilerplate but we don't need to worry about panics in return. 

Like I said, no need to log error if you're 100% sure it's always map. Just skipping is okay in that case. Please add a comment though.